src/lang/modelingWorkflows.ts

```diff
@@ -52,15 +53,24 @@ export async function updateModelingState(
   },
   options?: {
     focusPath?: Array<PathToNode>
+    skipUpdateAst?: boolean
   }
 ): Promise<void> {
-  // Step 1: Update AST without executing (prepare selections)
-  const updatedAst = await dependencies.kclManager.updateAst(
-    ast,
-    // false == mock execution. Is this what we want?
-    false, // Execution handled separately for error resilience
-    options
-  )
+  let updatedAst: {
+    newAst: Node<Program>
+    selections?: Selections
+  } = { newAst: ast }
+  // TODO: understand why this skip flag is needed for insertAstMod.
+  // It's unclear why we double casts the AST
```

Comment:
This took a lot of time from a few of us to debug. Thanks a lot to @franknoirot @nadr0 @lee-at-zoo-corp 

The solution below is not perfect but helps unblocking us without changing the code path too much.